pkg/config/config.go

```diff
@@ -30,38 +50,125 @@ const (
+			}),
+
+			Transform: func(obj interface{}) (interface{}, error) {
+				pod := obj.(*corev1.Pod)
+
+				newPod := &corev1.Pod{
+					TypeMeta: pod.TypeMeta,
+					ObjectMeta: v1.ObjectMeta{
+						Namespace: pod.Namespace,
+						Name:      pod.Name,
+						Labels:    pod.Labels,
+					},
+					Spec: corev1.PodSpec{
+						NodeName: pod.Spec.NodeName,
+					},
+				}
+
+				return newPod, nil
+			},
+		}
+
+		// Also for the profiles feature, we need to list the nodes, but we're only
+		// interested in the node name and the labels.
+		// Note that if in the future we need to list or get pods or nodes and use other
+		// fields we'll need to modify this function.
+		//
+		// Node being non-namespace resources shouldn't have a namespace configuration.
+		byObject[&corev1.Node{}] = cache.ByObject{
+			Transform: func(obj interface{}) (interface{}, error) {
+				node := obj.(*corev1.Node)
+
+				newNode := &corev1.Node{
+					TypeMeta: node.TypeMeta,
+					ObjectMeta: v1.ObjectMeta{
+						Name:   node.Name,
+						Labels: node.Labels,
+					},
+				}
+
+				return newNode, nil
+			},
+		}
+	}
+
+	return cache.Options{
+		DefaultNamespaces: getWatchNamespacesFromEnv(logger, watchNamespaceEnvVar),
+		ByObject:          byObject,
+	}
 }
 
-// ManagerOptionsWithNamespaces returns an updated Options with namespaces information.
-func GetNamespaceConfigs(logger logr.Logger, cacheConfig cache.Config) map[string]cache.Config {
-	namespaces := GetWatchNamespaces()
-	switch {
-	case len(namespaces) == 1:
-		logger.Info("Manager will be watching namespace", "namespace", namespaces[0])
-		return map[string]cache.Config{namespaces[0]: cacheConfig}
-	case len(namespaces) > 1:
-		// configure cluster-scoped with MultiNamespacedCacheBuilder
-		logger.Info("Manager will be watching multiple namespaces", "namespaces", namespaces)
-		// https://github.com/kubernetes-sigs/controller-runtime/commit/3e35cab1ea29145d8699259b079633a2b8cfc116
-		nsConfigs := make(map[string]cache.Config)
-		for _, ns := range namespaces {
-			nsConfigs[ns] = cacheConfig
+func getWatchNamespacesFromEnv(logger logr.Logger, envVar string) map[string]cache.Config {
+	cacheConfig := cache.Config{}
+
+	nsEnvValue, found := os.LookupEnv(envVar)
+	if !found {
+		logger.Info(fmt.Sprintf("CRD specific cache namespaces env variable %s not set, will be using common config", envVar))
+		nsEnvValue, found = os.LookupEnv(watchNamespaceEnvVar)
+		if !found {
+			logger.Info("Common namespace env variable %s not set, will be watching all namespaces", watchNamespaceEnvVar)
+			return map[string]cache.Config{cache.AllNamespaces: cacheConfig}
 		}
-		return nsConfigs
 	}
-	logger.Info("Manager will watch and manage resources in all namespaces")
-	return map[string]cache.Config{cache.AllNamespaces: cacheConfig}
+
+	var namespaces []string
+	if strings.Contains(nsEnvValue, ",") {
+		namespaces = strings.Split(nsEnvValue, ",")
+	} else {
+		namespaces = []string{nsEnvValue}
+	}
```

Comment:
could we simplify this to 
```suggestion
	namespaces := strings.Split(nsEnvValue, ",")
```
?

Reply:
Thanks!